### PR TITLE
Create friends

### PR DESCRIPTION
**Pull request by msmykowski**

### CODE REVIEW

**Comment by msmykowski** [comment]
testing ci

File: apps/api/lib/api/web/controllers/friendship_controller.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
+defmodule API.Web.FriendshipController do
+  use API.Web, :controller
+
+  action_fallback API.Web.FallbackController
+
+  def create(conn, %{"requestee_id" => requestee_id}) do
```
Comment (msmykowski) [inline apps/api/lib/api/web/controllers/friendship_controller.ex]: @camirmas do you think we should pass in the user_id from the client and save ourselves the user lookup, or just always rely on the token for user information?

File: apps/db/priv/repo/migrations/20170818010821_create_friendship.exs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+defmodule DB.Repo.Migrations.CreateFriendship do
+  use Ecto.Migration
+
+  def change do
+    create table(:friendships) do
+      add :requester_id, references(:users, on_delete: :delete_all)
```
Comment (msmykowski) [inline apps/db/priv/repo/migrations/20170818010821_create_friendship.exs]: If you have better names, let me know haha

File: apps/api/lib/api/web/controllers/fallback_controller.ex [hunk 3]
```diff
@@ -0,0 +1,18 @@
+defmodule API.Web.FallbackController do
+  use Phoenix.Controller
+
+  def call(conn, {:error, %Ecto.Changeset{} = changeset}), do: render_changeset_errors(conn, changeset)
+  def call(conn, {:error, _, %Ecto.Changeset{} = changeset, _}), do: render_changeset_errors(conn, changeset)
+
+  def unauthenticated(conn, _params) do
```
Comment (camirmas) [inline apps/api/lib/api/web/controllers/fallback_controller.ex]: :+1:

File: apps/api/test/api/web/controllers/friendship_controller_test.exs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+defmodule API.Web.FriendRequestControllerTest do
```
Comment (camirmas) [inline apps/api/test/api/web/controllers/friendship_controller_test.exs]: this file is still called `friendship_controller_test`

File: apps/db/lib/friend_request/friend_request.ex [hunk 5]
```diff
@@ -0,0 +1,40 @@
+defmodule DB.FriendRequest do
```
Comment (camirmas) [inline apps/db/lib/friend_request/friend_request.ex]: Sorry for not getting to this sooner, I'm curious about the decision to call it a `FriendRequest`. Do we anticipate ever having more info on this table that might make `Friendship` a more descriptive name? I think it's pretty clear either way, but something to think about

File: apps/db/lib/friend_request/friend_request.ex [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  defp changeset(struct, params) do
+    struct
+    |> cast(params, @fields)
+    |> validate_required(@fields)
+    |> assoc_constraint(:requesting_user)
+    |> assoc_constraint(:requested_user)
+    |> validate_inclusion(:status, @accepted_statuses)
+    |> unique_constraint(:requester_requestee, [name: :requester_requestee])
```
Comment (camirmas) [inline apps/db/lib/friend_request/friend_request.ex]: :+1:

File: apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  def change do
+    create table(:friend_requests) do
+      add :requesting_user_id, references(:users, on_delete: :delete_all)
+      add :requested_user_id, references(:users, on_delete: :delete_all)
+      add :status, :string
+
+      timestamps()
+    end
+
+    create unique_index(:friend_requests, [:requesting_user_id, :requested_user_id], name: :requester_requestee)
```
Comment (camirmas) [inline apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs]: this doesn't care about order right?

File: apps/api/lib/api/web/controllers/game_controller.ex [hunk 8]
```diff
@@ -2,6 +2,8 @@ defmodule API.Web.GameController do
   use API.Web, :controller
   alias Ecto.Multi
 
+  action_fallback API.Web.FallbackController
```
Comment (camirmas) [inline apps/api/lib/api/web/controllers/game_controller.ex]: :+1:

File: apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  def change do
+    create table(:friend_requests) do
+      add :requesting_user_id, references(:users, on_delete: :delete_all)
+      add :requested_user_id, references(:users, on_delete: :delete_all)
+      add :status, :string
+
+      timestamps()
+    end
+
+    create unique_index(:friend_requests, ["greatest(requesting_user_id,requested_user_id)", "least(requesting_user_id,requested_user_id)"], name: :requester_requestee)
```
Comment (msmykowski) [inline apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs]: what the fuck is this shit

File: apps/db/test/friend_request_test.exs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+      refute changeset.valid?
+    end
+
+    test "with creating the same friend_request relationship twice is invalid" do
+      {:ok, user_one} = User.create(@user_one_params)
+      {:ok, user_two} = User.create(@user_two_params)
+
+      {:ok, _friend_request} = FriendRequest.create(user_one.id, user_two.id, "pending")
+      {:error, changeset} = FriendRequest.create(user_one.id, user_two.id, "pending")
+      {:error, _changeset} = FriendRequest.create(user_two.id, user_one.id, "pending")
```
Comment (msmykowski) [inline apps/db/test/friend_request_test.exs]: extra check added here

File: apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs (hunk 9 above)
Comment (camirmas) [inline apps/db/priv/repo/migrations/20170818010821_create_friend_requests.exs]: ```¯\_(ツ)_/¯```

**Pull request merged**